shared/src/metabase/shared/util/internal/time.clj

```diff
@@ -66,39 +66,57 @@
 (defn- minus-ms [value]
   (t/minus value (t/millis 1)))
 
+(defn- apply-offset
+  [value offset-n offset-unit]
+  (t/plus
+    value
+    (case offset-unit
+      :minute (t/minutes offset-n)
+      :hour (t/hours offset-n)
+      :day (t/days offset-n)
+      :week (t/weeks offset-n)
+      :month (t/months offset-n)
+      :year (t/years offset-n)
+      (t/minutes 0))))
+
 (defmethod common/to-range :default [value _]
   ;; Fallback: Just return a zero-width at the input time.
   ;; This mimics Moment.js behavior if you `m.startOf("unknown unit")` - it doesn't change anything.
   [value value])
 
-(defmethod common/to-range :minute [value _]
-  (let [start (t/truncate-to value :minutes)]
-    [start (minus-ms (t/plus start (t/minutes 1)))]))
+(defmethod common/to-range :minute [value {:keys [n] :or {n 1}}]
+  (let [start (-> value
+                  (t/truncate-to :minutes))]
+    [start (minus-ms (t/plus start (t/minutes n)))]))
 
-(defmethod common/to-range :hour [value _]
-  (let [start (t/truncate-to value :hours)]
-    [start (minus-ms (t/plus start (t/hours 1)))]))
+(defmethod common/to-range :hour [value {:keys [n] :or {n 1}}]
+  (let [start (-> value
+                  (t/truncate-to :hours))]
+    [start (minus-ms (t/plus start (t/hours n)))]))
 
-(defmethod common/to-range :day [value _]
-  (let [start (t/truncate-to value :days)]
-    [start (minus-ms (t/plus start (t/days 1)))]))
+(defmethod common/to-range :day [value {:keys [n] :or {n 1}}]
+  (let [start (-> value
+                  (t/truncate-to :days))]
+    [start (minus-ms (t/plus start (t/days n)))]))
 
-(defmethod common/to-range :week [value _]
+(defmethod common/to-range :week [value {:keys [n] :or {n 1}}]
   (let [first-day (first-day-of-week)
         start (-> value
                   (t/truncate-to :days)
                   (t/adjust :previous-or-same-day-of-week first-day))]
-    [start (minus-ms (t/plus start (t/weeks 1)))]))
+    [start (minus-ms (t/plus start (t/weeks n)))]))
 
-(defmethod common/to-range :month [value _]
-  (let [value (t/truncate-to value :days)]
-    [(t/adjust value :first-day-of-month)
-     (minus-ms (t/adjust value :first-day-of-next-month))]))
+(defmethod common/to-range :month [value {:keys [n] :or {n 1}}]
+  (let [value (-> value
+                  (t/truncate-to :days)
+                  (t/adjust :first-day-of-month))]
+    [value (minus-ms (t/plus value (t/months n)))]))
 
-(defmethod common/to-range :year [value _]
-  (let [value (t/truncate-to value :days)]
-    [(t/adjust value :first-day-of-year)
-     (minus-ms (t/adjust value :first-day-of-next-year))]))
+(defmethod common/to-range :year [value {:keys [n] :or {n 1}}]
+  (let [value (-> value
+                  (t/truncate-to :days)
+                  (t/adjust :first-day-of-year))]
+    [value (minus-ms (nth (iterate #(t/adjust % :first-day-of-next-year n) value) n))]))
```

Comment:
I would expect `[value (minus-ms (t/plus value (t/years n)))]` to achieve the same.